File: jar-auth/src/main/java/com/beancounter/auth/client/AuthClientConfig.java
package com.beancounter.auth.client;

import com.beancounter.auth.common.TokenService;
import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
import org.springframework.boot.autoconfigure.http.HttpMessageConvertersAutoConfiguration;
import org.springframework.cloud.openfeign.EnableFeignClients;
import org.springframework.cloud.openfeign.FeignAutoConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

@Configuration
@Import({
    LoginService.class,
    AuthBeans.class,
    TokenService.class})

@EnableFeignClients(basePackages = "com.beancounter.auth")
@ImportAutoConfiguration({
    HttpMessageConvertersAutoConfiguration.class,
    FeignAutoConfiguration.class})

public class AuthClientConfig {
}

File: jar-auth/src/main/java/com/beancounter/auth/client/AuthClientConfig.kt
package com.beancounter.auth.client

import com.beancounter.auth.common.TokenService
import org.springframework.boot.autoconfigure.ImportAutoConfiguration
import org.springframework.boot.autoconfigure.http.HttpMessageConvertersAutoConfiguration
import org.springframework.cloud.openfeign.FeignAutoConfiguration
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import

@Configuration
@Import(LoginService::class, AuthBeans::class, TokenService::class)
@ImportAutoConfiguration(
    HttpMessageConvertersAutoConfiguration::class, FeignAutoConfiguration::class
)
class AuthClientConfig
